libV2/schemaUtils.js

```diff
@@ -2283,13 +2480,26 @@ let QUERYPARAM = 'query',
     }
 
     _.forOwn(operationItem.responses, (responseObj, code) => {
-      let responseSchema = _.has(responseObj, '$ref') ?
-          resolveSchema(context, responseObj, { isResponseSchema: true }) : responseObj,
+      let responseSchema = _.has(responseObj, '$ref') ? (
+          resolveSchema(context, responseObj, { isResponseSchema: true, isBodySchema: true })) : responseObj,
         { includeAuthInfoInExample } = context.computedOptions,
         auth = request.auth,
-        resolvedExamples = resolveResponseBody(context, responseSchema, requestBodyExamples, code) || {},
-        headers = resolveResponseHeaders(context, responseSchema.headers);
-
+        resolvedExamples = resolveResponseBody(context, responseSchema, requestBodyExamples, code) || {};
+      resolvedExamplesList = context.resolvedSchemaTypes;
+      context.resolvedSchemaTypes = null;
+      // eslint-disable-next-line one-var
+      let headers = resolveResponseHeaders(context, responseSchema.headers),
+        bodyHeaderObj,
+        respBlock,
+        resolvedHeadersList = context.resolvedSchemaTypes;
```

Comment:
At line [2488](https://github.com/postmanlabs/openapi-to-postman/pull/834/files#diff-d2a91c8370a4bdef207a0c7fd03f4e03d928c4baa87b0768d51093480c71f077R2488) you're using `context.resolvedSchemaTypes` as example list. But here you're using that as headerlist. That seems very counter intuitive 😞